### ISSUE TITLE
Release 1.14

### ISSUE THREAD

**Issue by Nyholm** [php-http/HttplugBundle#296]
I think we should prepare for a new minor release. 

@mponos mentioned in https://github.com/php-http/HttplugBundle/pull/295#pullrequestreview-188449117 that we should do a major release. Could you please elaborate?

**Comment by gmponos** [comment]
I created a test application with this bundle. Except if I did something wrong in order to inject a PSR client I had to use

```
        "php-http/guzzle6-adapter": "^2.0",
        "php-http/client-common": "2.x-dev",
```

both of the packages above drop sup for PHP 5. In the end you will have a Bundle which will be universal for both of `php-http/*: v1` and `php-http/*: v2`.. and you will try to support both of these... 

The BC for the consumers are already covered when they will try to use for instance `guzzle6-adapter:v2`

Their upgrade path is easy.. so in the end the maintenance cost that I see here is bigger than it is worth it..

**Comment by gmponos** [comment]
> which will be universal for both of php-http/*: v1 and php-http/*: v2

if this is what you guys want then there is nothing wrong with this.. My personal preference in general is that it is better to bump to a new version.

**Comment by gmponos** [comment]
> The issue you mention here and on slack is that you cannot use HTTPlug 1.0 together with a PSR-18 client. 

Yes exactly.. I though that your the goal of your [PR](https://github.com/php-http/HttplugBundle/pull/295) initially was that.. personally I am not 100% that it should... I can admit that it offers more BC compatibility but at the same time I find it a little bit overkill.

Let's brake it down... let's say that I want to do that and use HTTPlug 1.0 with PSR-18.. 
First of all this means that I am already on PHP 7.

Secondly what reason do I have not to update my packages to v2... 

HTTPlug client, guzzle-adapter, and common package have minimum changes

https://github.com/php-http/httplug/compare/v1.1.0...v2.0.0

The only valid reason that I can think of is if a consumer implemented his own HTTPlug Client. If we force him to move to v2 then we force him to implement PSR-18.

**Comment by Nyholm** [comment]
> Yes exactly.. I though that the goal of your PR initially was that 

It wasn't but it is a good goal. 

--------

I know the changes are minimal between 1.x and 2.x. But the BC layer here is minimal. The major changes happend in client-common. With php-http/client-common#143 that is all in place. Then we drop most of it when we release php-http/client-common:2.0. 

Bumping this bundle to 2.0 will only make the upgrade path harder. I want to keep as many things 1.0 as possible. 
If we want to change some major thing or release 2.0 in 6 months, sure, lets talk then.

**Comment by Nyholm** [comment]
We are currently blocked by 

php-http/client-common: 1.9
php-http/stopwatch-plugin: 1.2
php-http/cache-plugin: 1.6
php-http/logger-plugin: 1.1

**Comment by dbu** [comment]
i agree with @Nyholm. the problem with doing a version 2 of this is that people stuck on php 5 or otherwise not willing to upgrade either get no updates anymore, or we have to maintain two versions of the bundle. as the overhead for staying BC is minimal, i think its better to stay version 1 for now.

when there is a need in this bundle to bump to php 7.1 to use some of the new features of php, i am happy to go to a version 2. but as tobias says, thats something for later.
